src/Native/Unix/System.Native/pal_termios.cpp

```diff
@@ -0,0 +1,360 @@
+    case B600:
+        return 600;
+    case B300:
+        return 300;
+    case B200:
+        return 200;
+    case B150:
+        return 150;
+    case B134:
+        return 134;
+    case B110:
+        return 110;
+    case B75:
+        return 75;
+    case B50:
+        return 50;
+    }
+
+    return (static_cast<int>(brate));
+}
+
+extern "C" int32_t
+SystemNative_TermiosGetSpeed(int fd)
+{
+    struct termios       term;
+    if (tcgetattr(fd, &term) < 0)
+    {
+        return  -1;
+    }
+
+    return SystemNative_TermiosRate2speed(cfgetispeed(&term));
+}
+
+extern "C" int32_t
+SystemNative_TermiosSetSpeed(int fd, int speed)
+{
+    struct termios       term;
+    if (tcgetattr(fd, &term) < 0)
+    {
+        return  -1;
+    }
+
+    cfsetspeed(&term, SystemNative_TermiosSpeed2rate(speed));
+
+    if (tcsetattr(fd, TCSANOW, &term) < 0)
+    {
+        return -2;
+    }
+
+    return speed;
+}
+
+extern "C" int32_t
+SystemNative_TermiosAvailableBytes(int fd, int readBuffer)
+{
+    int32_t bytes;
+    if (ioctl (fd, readBuffer ? FIONREAD : TIOCOUTQ, &bytes) == -1) {
+        return -1;
+    }
+
+    return bytes;
+}
+
+extern "C" int32_t
+SystemNative_TermiosDiscard(int fd, int queue)
+{
+    return tcflush(fd, queue == 0 ? TCIOFLUSH : queue == 1 ? TCIFLUSH : TCOFLUSH);
+}
+
+extern "C" int32_t
+SystemNative_TermiosDrain(int fd)
+{
+    return tcdrain(fd);
+}
+
+extern "C" int32_t
+SystemNative_TermiosSendBreak(int fd, int duration)
+{
+    return tcsendbreak(fd, duration);
+}
+
+extern "C" int32_t
+SystemNative_TermiosReset(int fd, int speed, int dataBits, int stopBits, int parity, int handshake)
+{
+    struct termios       term;
+    int ret = 0;
+
+    if (tcgetattr(fd, &term) < 0)
+    {
+        return  -1;
+    }
+
+    cfmakeraw(&term);
+    term.c_cflag |=  (CLOCAL | CREAD);
+    term.c_lflag &= ~(static_cast<tcflag_t>(ICANON | ECHO | ECHOE | ECHOK | ECHONL | ISIG | IEXTEN ));
+    term.c_oflag &= ~(static_cast<tcflag_t>(OPOST));
+    term.c_iflag = IGNBRK;
+
+    term.c_cflag &= ~(static_cast<tcflag_t>(CSIZE));
+    switch (dataBits)
+    {
+    case 5:
+        term.c_cflag |= CS5;
+        break;
+    case 6:
+        term.c_cflag |= CS6;
+        break;
+    case 7:
+        term.c_cflag |= CS7;
+        break;
+    case 8:
+    default:
+        term.c_cflag |= CS8;
+        break;
+    }
+
+    // There really is only choice of 1 or 2.
+    if (stopBits == 1)
+    {
+        term.c_cflag &= ~(static_cast<tcflag_t>(CSTOPB));
+    }
+    else if (stopBits > 1)
+    {
+        term.c_cflag |= CSTOPB;
+    }
+
+    // Set parity
+    term.c_iflag &= ~(static_cast<tcflag_t>(INPCK | ISTRIP));
+    switch (parity)
+    {
+    case ParityOdd:
+        term.c_cflag |= PARENB | PARODD;
+        break;
+    case ParityEven:
+        term.c_cflag |= PARENB;
+        term.c_cflag &= ~(static_cast<tcflag_t>(PARODD));
+        break;
+    default:
+        term.c_cflag &= ~(static_cast<tcflag_t>(PARENB | PARODD));
```

Comment:
A nit - here you clear the parity bits in the default and in the code below, you pre-clear the xon/xoff and rts/cts  bits. It would be nice to be consistent.